testcrate/ui/not-a-result.stderr

```diff
@@ -18,14 +27,31 @@ error[E0277]: the trait bound `u32: futures::__rt::IsResult` is not satisfied
    |
    = help: the trait `futures::__rt::IsResult` is not implemented for `u32`
 
+error[E0308]: mismatched types
+  --> $DIR/not-a-result.rs:13:13
+   |
+13 | fn bar() -> u32 {
+   |             ^^^ expected u32, found enum `std::result::Result`
```

Comment:
Hm this seems like it may be a regression? This error could be sort of confusing...